Doc/howto/descriptor.rst

```diff
@@ -1004,47 +1004,50 @@ here is a pure Python equivalent:
             if doc is None and fget is not None:
                 doc = fget.__doc__
             self.__doc__ = doc
-            self._name = ''
+            self.__name__ = None
 
         def __set_name__(self, owner, name):
-            self._name = name
+            self.__name__ = name
 
         def __get__(self, obj, objtype=None):
             if obj is None:
                 return self
             if self.fget is None:
                 raise AttributeError(
-                    f'property {self._name!r} of {type(obj).__name__!r} object has no getter'
+                    f'property {self.__name__!r} of {type(obj).__name__!r} '
+                    'object has no getter'
                  )
             return self.fget(obj)
 
         def __set__(self, obj, value):
             if self.fset is None:
                 raise AttributeError(
-                    f'property {self._name!r} of {type(obj).__name__!r} object has no setter'
+                    f'property {self.__name__!r} of {type(obj).__name__!r} '
+                    'object has no setter'
                  )
             self.fset(obj, value)
 
         def __delete__(self, obj):
             if self.fdel is None:
                 raise AttributeError(
-                    f'property {self._name!r} of {type(obj).__name__!r} object has no deleter'
+                    f'property {self.__name__!r} of {type(obj).__name__!r} '
+                    'object has no deleter'
                  )
             self.fdel(obj)
 
         def getter(self, fget):
             prop = type(self)(fget, self.fset, self.fdel, self.__doc__)
-            prop._name = self._name
+            prop.__name__ = self.__name__
             return prop
 
         def setter(self, fset):
             prop = type(self)(self.fget, fset, self.fdel, self.__doc__)
-            prop._name = self._name
+            prop.__name__ = self.__name__
```

Comment:
I am not sure that this is correct, but this is a different issue.

Reply:
This is mostly useful when constructing properties manually outside of the class, which is advanced/unusual. I think it is harmless. Anyway, this is the logic in the C code which has been there since prop_name was first introduced.

Reply:
I mean that something like `prop.__name__ = fset.__name__` or `prop.__name__ = getattr(fset, '__name__', self.__name__)` could be more expected. But this is a different issue.